core/src/main/java/io/fabric8/maven/core/handler/ContainerHandler.java

```diff
@@ -104,7 +104,7 @@ private String getImageName(ImageConfiguration imageConfiguration) {
             project.getProperties().getProperty("docker.registry"));
 
         String prefix = "";
-        if (StringUtils.isNotBlank(configuredRegistry)) {
+        if (StringUtils.isNotBlank(configuredRegistry) && !imageConfiguration.getName().contains(configuredRegistry)) {
```

Comment:
What is if `configuredRegistry` is not blank, but imageConfiguration's name contains also a registry *but* different to configuredRegistry ? I would the expect that the ImageConfiguration's registry would "wind".

Reply:
Sorry, but what does "wind" mean in this context? You mean we should give preference to configuredRegistry?

Reply:
sorry, meant "win". Fat fingers ;-) 

Yes, the registry closes to the specified image name should take precedence. This is what we use as a rule of thumb also in d-m-p.